tests/common/plugins/sanity_check/__init__.py

```diff
@@ -416,17 +374,35 @@ def recover_on_sanity_check_failure(duthosts, failed_results, fanouthosts, local
     new_failed_results = [result for result in new_check_results if result['failed']]
     if new_failed_results:
         request.config.cache.set(cache_key, True)
-        request.config.cache.set(recovery_cache_key, False)
+        add_custom_msg(request, f"dut_check_result.{cache_key}", True)
+        add_custom_msg(request, f"dut_check_result.{recovery_cache_key}", False)
         pt_assert(False,
                   f"!!!!!!!!!!!!!!!! {sanity_check_stage} sanity check after recovery failed: !!!!!!!!!!!!!!!!\n"
                   f"{json.dumps(new_failed_results, indent=4, default=fallback_serializer)}")
     # Record recovery success
-    request.config.cache.set(recovery_cache_key, True)
+    add_custom_msg(request, f"dut_check_result.{recovery_cache_key}", True)
+
+
+def add_custom_msg(request, key, val):
```

Comment:
add_custom_msg should be the function from tests/common/helpers/custom_msg_utils.py
So that it could be a shared utility.